### ISSUE TITLE
Framework: AT: No PRs merging due to potential communication issues and failing framework tests

### ISSUE THREAD

**Issue by GrahamBenHarper** [trilinos/Trilinos#10985]
## Bug Report
@trilinos/framework 

As reported in [TRILINOSHD-201](https://sems-atlassian-son.sandia.gov/jira/servicedesk/customer/portal/7/TRILINOSHD-201), several pull requests are experiencing framework issues. Specifically, there's the following communication issue in  #10983:
```
NOTICE: The AutoTester has encountered an internal error (usually a Communications Timeout), testing will be restarted, previous tests may still be running but will be ignored by the AutoTester...
```

After digging deeper, it's also clear that there are framework tests failing related to #10972, #10983, and more. Some output from the failing test is included below:
```

XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX

Advanced Test: TrilinosFrameworkTests_PullRequestLinuxDriver_UnitTests

Selected Test/CTest Properties:
  CATEGORIES = BASIC
  PROCESSORS = 1
  TIMEOUT    = DEFAULT

Creating new working directory '/scratch/trilinos/jenkins/ascic158/workspace/Trilinos_PR_python3/pull_request_test/commonTools/framework/TrilinosFrameworkTests_PullRequestLinuxDriver_UnitTests'

Running test commands: TEST_0

================================================================================

TEST_0

Running: "bash" "pr_python_test_driver.sh" "/projects/sems/install/rhel7-x86_64/sems/v2/utility/anaconda3/2020.07/gcc/7.3.0/7t6phgv/bin/python3" "pip3"

  Running in working directory "/scratch/trilinos/jenkins/ascic158/workspace/Trilinos_PR_python3/Trilinos/commonTools/framework/../../packages/framework/pr_tools"

--------------------------------------------------------------------------------


+----------------------------------------------------------+
|           P Y T H O N   T E S T   D R I V E R            |
|                                                          |
|                   2022-09-02 16:25:28                    |
+----------------------------------------------------------+


+----------------------------------------------------------+
|                P A C K A G E   S E T U P                 |
|                                                          |
|                   2022-09-02 16:25:28                    |
+----------------------------------------------------------+
--- python: /projects/sems/install/rhel7-x86_64/sems/v2/utility/anaconda3/2020.07/gcc/7.3.0/7t6phgv/bin/python3
--- pip   : pip3
PRDriver> pip executable: /projects/sems/install/rhel7-x86_64/sems/v2/utility/anaconda3/2020.07/gcc/7.3.0/7t6phgv/bin/python3 -m pip
PRDriver> 
PRDriver> /projects/sems/install/rhel7-x86_64/sems/v2/utility/anaconda3/2020.07/gcc/7.3.0/7t6phgv/bin/python3 -m pip install --user configparser
Requirement already satisfied: configparser in /home/trilinos/.local/lib/python3.8/site-packages (5.3.0)
/scratch/trilinos/jenkins/ascic158/workspace/Trilinos_PR_python3/Trilinos/packages/framework/pr_tools/common.bash: line 78: color: command not found
PRDriver> OK (3 s)


+----------------------------------------------------------+
|                E X E C U T E   T E S T S                 |
|                                                          |
|                   2022-09-02 16:25:31                    |
+----------------------------------------------------------+
--- /projects/sems/install/rhel7-x86_64/sems/v2/utility/anaconda3/2020.07/gcc/7.3.0/7t6phgv/bin/python3 -m pytest --cov=. --cov-report term-missing
ERROR: usage: __main__.py [options] [file_or_dir] [file_or_dir] [...]
__main__.py: error: unrecognized arguments: --cov=. --cov-report
  inifile: None
  rootdir: /scratch/trilinos/jenkins/ascic158/workspace/Trilinos_PR_python3/Trilinos/packages/framework/pr_tools


--- STATUS: 4

--------------------------------------------------------------------------------

TEST_0: Return code = 4
TEST_0: Pass criteria = Match any REGEX {=+ [0-9]+ passed in} [not matched]
TEST_0: Result = FAILED

================================================================================

OVERALL FINAL RESULT: TEST FAILED (TrilinosFrameworkTests_PullRequestLinuxDriver_UnitTests)

XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
```

**Comment by jhux2** [comment]
> Take a look at [this query](https://trilinos-cdash.sandia.gov/queryTests.php?project=Trilinos&begin=2022-09-01&end=2022-09-06&filtercount=2&showfilters=1&filtercombine=and&field1=groupname&compare1=61&value1=Pull%20Request&field2=testname&compare2=61&value2=TrilinosFrameworkTests_PullRequestLinuxDriver_UnitTests) and [this query](https://trilinos-cdash.sandia.gov/index.php?project=Trilinos&begin=2022-09-01&end=2022-09-06&filtercount=2&showfilters=1&filtercombine=and&field1=groupname&compare1=61&value1=Pull%20Request&field2=buildname&compare2=63&value2=rhel7_sems-gnu-7.2.0-anaconda3-serial_debug_shared_no-kokkos-arch_no-asan_no-complex_no-fpic_no-mpi_no-pt_no-rdc_no-uvm_deprecated-on_pr-framework). These failures have nothing to do the particular PR being tested and every PR test iteration will fail until someone does something to fix this failing test `TrilinosFrameworkTests_PullRequestLinuxDriver_UnitTests`.

@bartlettroscoe Clicking through these two links shows exactly the error message in the description.

**Comment by jmlapre** [comment]
They are failing because the color command can't be found.  If you notice in the script (common.bash) it fails due to color not being found after it installs configparser.  Then it stops loading the rest of the packages and that's why the coverage is failing (it hasn't installed pytest-cov due to the earlier failure).

**Comment by jhux2** [comment]
Might be fixed by #10994.

**Comment by jhux2** [comment]
This is fixed.  [Dashboard results](https://trilinos-cdash.sandia.gov/index.php?project=Trilinos&filtercount=5&showfilters=1&filtercombine=and&field1=buildname&compare1=63&value1=PR-&field2=buildstarttime&compare2=83&value2=2022-09-05&field3=buildstarttime&compare3=84&value3=2022-09-10&field4=groupname&compare4=61&value4=Pull%20Request&field5=label&compare5=61&value5=TrilinosFrameworkTests) before and after merge of #10994.
